pkg/sql/catalog/tabledesc/table_desc.go

```diff
@@ -143,6 +143,16 @@ func (desc *wrapper) IsTemporary() bool {
 	return desc.GetTemporary()
 }
 
+// Hydrated implements the TableDescriptor interface.
+func (desc *wrapper) Hydrated() bool {
+	for _, col := range desc.VisibleColumns() {
```

Comment:
Why `VisibleColumns` and not `AllColumns`?

Reply:
In fact do we need any of this when we already have `ContainsUserDefinedTypes` in `catalog.HydratableDescriptor`?

Reply:
> Why VisibleColumns and not AllColumns?

because we only look at visible columns when constructing the implicit record type.

> In fact do we need any of this when we already have ContainsUserDefinedTypes in catalog.HydratableDescriptor?

`ContainsUserDefinedTypes` doesn't check if types are hydrated or not :(

Reply:
Thanks!